deployment/jenkins/azureutil.groovy

```diff
@@ -66,36 +66,37 @@ def prepareEnv(String targetEnv) {
 
 def deployFunction() {
     sh """
-        # Storage connection for images 
+        # Storage connection for images
         storage_name=\$(az storage account list -g ${config.COMMON_GROUP} --query [2].name | tr -d '"')
         storage_conn_str=\$(az storage account show-connection-string -g ${config.COMMON_GROUP} -n \${storage_name} --query connectionString | tr -d '"')
-        
+
         function_id=\$(az functionapp list -g ${config.COMMON_GROUP} --query [0].id | tr -d '"')
         az functionapp config appsettings set --ids \${function_id} --settings STORAGE_CONNECTION_STRING=\${storage_conn_str}
         az functionapp deployment source sync --ids \${function_id}
     """
 }
 
-def deployWebApp(String resGroup) {
+def deployWebApp(String resGroup, String dockerFilePath) {
+    def appName = sh(
+            script: "az webapp list -g ${resGroup} --query [0].name | tr -d '\"'",
+            returnStdout: true
+    ).trim()
+
+    azureWebAppPublish appName: appName, azureCredentialsId: 'azure-sp', dockerFilePath: dockerFilePath, dockerImageName: '', dockerImageTag: '', dockerRegistryEndpoint: [credentialsId: 'acr', url: "https://${this.acrName}.azurecr.io"], filePath: '', publishType: 'docker', resourceGroup: resGroup, slotName: '', sourceDirectory: '', targetDirectory: ''
```

Comment:
So credential id for ACR is still required, right?
Since difference environments (dev, test, prod) have different ACRs, does it work for different environments?

Reply:
Instead of ACR username/password, now I use service principal to access ACR. Looks like you use a single service principal for all there environments so I guess it will work too?

Reply:
Cool! Let me do a test run before merge.

Reply:
I just had a test run. Seems that the image name is `NGINX` here? Can we change it to `web-app` to align with rest of the project.

Reply:
Yes, I forgot there is an option `dockerImageName`, Updated.